### ISSUE TITLE
Replace the current biome colors for the AMIDST list

### ISSUE THREAD

**Issue by vpontin** [gegy1000/EarthBiomemap#4]
Hi! First ticket after coming to Curseforge asking about biome fixes! Before starting to make the fixes for South American continent, i find the biome list is lacking several key biomes for my work.

I know this issue is already discussed in #2 and #3 , but i think my request will only mess with this tickets and need to be standalone. 

So, the request is to "sync" our color list with [AMIDST tool](https://github.com/toolbox4minecraft/amidst) list.

I know this question seems to be ridiculous, but using AMIDIST list instead of the current custom one would be better because AMIDST is actively developed, and tracking its biome changes would facilitate the addition of newer Minecraft biome additions.

I additionally create a [companion ticket](https://github.com/toolbox4minecraft/amidst/issues/207) on AMIDST repo to make sure the author allow me to update our color list.

[The list](https://github.com/toolbox4minecraft/amidst/blob/master/biome/default.json) on AMIDST is stored in JSON format, while [our list](https://github.com/gegy1000/EarthBiomemap/blob/master/Color%20to%20Biome%20Map.txt) is stored on plain TXT with the colors encoded on hex format. I think about create a script (maybe in Lazarus or Javascript) to do the dirty work of conversion.

And yes, i know this would mean we need to recolor all biome images, but i'm willing to work on it (i just need to find a better tool to make the color replace safe, Photoshop is very tricky in this case).

The only technical caveat would be if AMIDST's GPL would interfere in some way, anyway i asked this on the companion ticket.

**Comment by gegy1000** [comment]
I like the idea.

This list is just a reference for people contributing to the biomemap, I don't actually read from that file.

**Comment by vpontin** [comment]
So if i recolor all images you will need to change the colors in the code right?

Or you just use the images (and colors) to reference?

**Comment by vpontin** [comment]
I'm currently ma . Just waiting the another issue on AMIDST repo to PR here (I think i learned how to do this).

Also i added an ACO swatches file, with all the colors in the new txt list. The swatches have the biome colors ordered by Biome IDS, and a caveat is if you import the swatches as the colors list of the image (in photoshop) the colors will mess up (because the original order in each image is different) and it will require a recolor. Again, i dont know if internal things like this (color order in index) are important to your code, if not, better.

As i said above, most of the images are color indexed, something that greatly facilitated the work. Sadly, my PS is a joke and dont handle 24-bit/channel pngs, so all the images would be stored in 8-bits/channel. If this would hurt the code let me know.

There is A LOT of unlisted colors in some regions of the globe (like Norway's Fjords), and i'm trying to figure out what biome they represent, or if they are artifacts. Any help to figure out this would be appreciated ^^

Another thing i noted was some biomes in AMIDST list dont exist according to [MC Wiki data values](http://minecraft.gamepedia.com/Biome#Biome_IDs) (a headache i didnt predicted when i requested that, sorry), so it would require a cleanup of both txt and ACO lists before PR.

Anyway, the work is almost done. Just i need to figure what i need to do with the extra colors and mate!

**Comment by vpontin** [comment]
Any thoughts @gegy1000 ???

**Comment by hagnat** [comment]
should we sync the map colors with AMIDST, this would resolve #2

**Comment by vpontin** [comment]
Exacly.

But i stopped the conversion because several images have colors outside the current list; i need to learn with @gegy1000 what is your opinion: should i replace the colors with the most "similar" biome in the current index, set to Ocean or whatever

**Comment by vpontin** [comment]
Another thing i was thinking was to "rearrange" the internal color index of the images, but it will require a lot of work (in photoshop) to rearrange the colors. But in the end it would be far more easy to maintain

**Comment by vpontin** [comment]
Updates from https://github.com/toolbox4minecraft/amidst/issues/207, i strongly recommend you read.

Well, it seems that even the AMIDST color palette is under GPL. This means that it may not be possible to use the colors here due to license incompatibility (but again: read the companion issue).

Its up to @gegy1000 if i continue with the AMIDST recolor or if you get back with @hagnat's #2 and use another colors.

**Comment by gegy1000** [comment]
The current biomemap will be redone because I am adding support for Biomes O' Plenty.

**Comment by hagnat** [comment]
so, are you including some sort of fallback, in case the player does not have BOP ? like, boreal forest becomes regular forest, lush desert becomes regular desert (or plateau)

**Comment by gegy1000** [comment]
Yes.

**Comment by vpontin** [comment]
So, my participation in this project ends here. I wanted to add more colors to the list to be able to fix the wrong biome placement at least in my home continent, but since @gegy1000 will add suport for BoP/EXL type of crap, i'm afrain i will be unable to do my work, since i will stick with vanilla biomes.

Anyway, closing it now since we get a response from AMIDST.
